src/variable_rate.jl

```diff
@@ -387,8 +408,10 @@ function (cache::VR_DirectEventCache)(u, t, integrator)
     return cache.current_threshold
 end
 
-function execute_affect!(vjumps, integrator, idx)
-    @inbounds vjumps[idx].affect!(integrator)
+@generated function execute_affect!(cache::VR_DirectEventCache{T, F1, F2, RNG}, integrator, idx) where {T, F1, F2, RNG}
+    quote
+        @inbounds Base.Cartesian.@nif $(fieldcount(F2)) i -> (i == idx) i -> (cache.affect_funcs[i](integrator)) i -> (cache.affect_funcs[fieldcount(F2)](integrator))
```

Comment:
```suggestion
        Base.Cartesian.@nif $(fieldcount(F2)) i -> (i == idx) i -> (@inbounds cache.affect_funcs[i](integrator)) i -> (@inbounds cache.affect_funcs[fieldcount(F2)](integrator))
```
I think the inbounds need to be on the commands that actually get called here.